phpBB/posting.php

```diff
@@ -620,9 +624,10 @@
 }
 unset($uninit);
 
+($attachment_poster_id !== '') ?: $attachment_poster_id = $post_data['poster_id'];
```

Comment:
Maybe you could initialize `$attachment_poster_id = $post_data['poster_id'];` and remove the above line? It would make the code a bit easier to follow and event handlers would be able to use the `attachment_poster_id` value for read-only purposes, too.

Reply:
I remember why I changed it to that now.  At first I did do what you suggested, though I realized that `$message_parser->get_submitted_attachment_data()` is calling the poster_id made after [the foreach on this line](https://github.com/phpbb/phpbb/blob/179fc6e7bc35c77c0b00ead381e0834e302316eb/phpBB/posting.php#L618).  This means that with your suggestion, if the event is unused, it will be checking it against the $post_data['poster_id'] variable before it is possibly changed.  This may lead to unintended behavior as it was originally coded to use the poster_id after all of that business is done.

I tried a few ways of going about this, but I settled on checking if the new attachment_poster_id has been used at all; if it has then we use that variable, if it hasn't then we use the $post_data['poster_id'] value that has potentially been updated after that foreach I linked.